ddcz/views/legacy.py

```diff
@@ -0,0 +1,135 @@
+#  - Anything from CREATION and CREATION_DETAIL
+#  - Anything from ALLOWED_CREATION_PAGES
+PAGE_TO_VIEW_MAP = {
+    "aktuality": "ddcz:news",
+    "novinky": "ddcz:newsfeed",
+    "uzivatele": "ddcz:users-list",
+    "registrace": "ddcz:sign-up",
+    "seznamka": "ddcz:dating",
+    "forum": "ddcz:phorum-list",
+    "linky": "ddcz:links-list",
+    "inzerce": "ddcz:market",
+    "credits": "ddcz:web-authors-and-editors",
+    "faq": "ddcz:website-manual",
+    "oao": "ddcz:faq",
+    "cojetodrd": "ddcz:about-drd",
+}
+
+COMMON_ARTICLES_NAME_MAP = {
+    "hrbitov": "hrbitov",
+    "clanky": "clanky",
+    "expanze": "expanze",
+    "kouzelnikzvl": "kouzelnikzvl",
+    "alchymistazvl": "alchymistazvl",
+    "valecnik": "valecnik",
+    "hranicar": "hranicar",
+    "zlodej": "zlodej",
+    "novapovolani": "novapovolani",
+    "noverasy": "noverasy",
+}
+
+
+ALLOWED_CREATION_PAGES = [
+    "bestiar",
+    "dobrodruzstvi",
+    "predmety",
+    "kouzla",
+    "alchpredmety",
+    "hranicarkouzla",
+    "dovednosti",
+    "galerie",
+    "fotogalerie",
+]
+
+
+@require_http_methods(["GET"])
+def legacy_router(request):
+
+    section = request.GET.get("rub", False)
+    subsection = request.GET.get("co", False)
+    id = request.GET.get("id", False)
+
+    # The LEGACY_PLAIN_ROUTER is redirecting basic pages.
+    # Typically no creative pages are present here.
+    if section in PAGE_TO_VIEW_MAP.keys():
+        return HttpResponseRedirect(reverse(PAGE_TO_VIEW_MAP[section]))
+
+    # Some of the creation have legacy url
+    # index.php?rub=prispevky(_komp)&co=page_slug
+    # Here are lists of such creative pages.
+    if section in CREATION and subsection in COMMON_ARTICLES_NAME_MAP.keys():
+        return HttpResponseRedirect(
+            reverse(
+                "ddcz:creation-list",
+                kwargs={"creative_page_slug": COMMON_ARTICLES_NAME_MAP[subsection]},
+            )
+        )
+
+    # For index.php?rub=prispevky_jeden&subsection=page_slug&id=article_id
+    # we can find the article detail by Id.
+    if section == CREATION_DETAIL and id is not False:
+        page = get_object_or_404(CreativePage, slug=subsection)
+        return get_creation_detail_redirect(page, id)
+
+    # There are some special creative pages that are not stored
+    # as the others, those have their own tables in the database.
+    # For those we have lists and details in this for loop.
+    for name in ALLOWED_CREATION_PAGES:
+        if section in [name, name + "_komp"]:
+            return HttpResponseRedirect(
+                reverse(
+                    "ddcz:creation-list",
+                    kwargs={"creative_page_slug": name},
+                )
+            )
+        if section in [name + "_jeden"]:
+            page = get_object_or_404(CreativePage, slug=name)
+            return get_creation_detail_redirect(page, id)
+
+    ###  Finally if no route is found, redirect to news
+    logger = logging.getLogger(__name__)
```

Comment:
I think this would better go to module-level as everywhere else

Reply:
Could you point me somewhere where it's used so I could see it pls? :smiley_cat:

Reply:
https://github.com/dracidoupe/graveyard/blob/master/ddcz/tavern.py#L31 :)

(the reason is that if we'd add more logging in the future, it's available and people don't create multiple instances unecessarily :) )

Reply:
I made a method in a special legacy.py file, I hope this is how you meant it. :) I also edited the message according to the next comment.

Reply:
No, what I mean is to put the logger on top of the current file to make it a module scope instead of the function where it's a function scoped. 

Maybe let's talk about it on a call.

Reply:
Agree, call can make it clearer for me :)